activesupport/test/json/encoding_test.rb

```diff
@@ -427,6 +427,17 @@ def test_exception_to_json
     assert_equal '"foo"', ActiveSupport::JSON.encode(exception)
   end
 
+  class DetailedException < Exception
+    def initialize(message)
+      @message = message
+    end
+  end
+
+  def test_exception_with_instance_values_to_json
+    exception = DetailedException.new("foo")
+    assert_equal %({"message":"foo"}), ActiveSupport::JSON.encode(exception)
+  end
+
```

Comment:
I'm open to suggestions, but the current behavior to always return a String seems strange to me. I'd like to return a JSON object, if there are keys and values for it.